clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockOutputStream.java

```diff
@@ -84,13 +87,27 @@ public KafkaLZ4BlockOutputStream(OutputStream out, int blockSize, boolean blockC
     /**
      * Create a new {@link OutputStream} that will compress data using the LZ4 algorithm.
      *
+     * @param out The output stream to compress
+     * @param blockSize Default: 4. The block size used during compression. 4=64kb, 5=256kb, 6=1mb, 7=4mb. All other
+     *            values will generate an exception
+     * @param blockChecksum Default: false. When true, a XXHash32 checksum is computed and appended to the stream for
+     *            every block of data
+     * @throws IOException
+     */
+    public KafkaLZ4BlockOutputStream(OutputStream out, int blockSize, boolean blockChecksum) throws IOException {
```

Comment:
Correct -- kafka doesn't currently use them, but we don't know whether users have written code against these interfaces. I prefer to think of these classes as general implementations of the LZ4 spec, somewhat independent of kafka's specific use.